src/util/sorter.js

```diff
@@ -4,28 +4,30 @@ import gbp from "../assets/images/gbp.png";
 import yen from "../assets/images/yen.png"
 
 export const transformData = (rawData) => {
-  const mapDataToObject =
-    rawData.reduce((acc, curr) => {
-      const accValue = acc[curr.node.date] || {
-        date: curr.node.date
-      };
+  if(!!rawData){
+    const mapDataToObject =
+      rawData.reduce((acc, curr) => {
+        const accValue = acc[curr.node.date] || {
+          date: curr.node.date
+        };
 
-      const currPeriod = curr.node.rate.periodOfDay.toLowerCase();
+        const currPeriod = curr.node.rate.periodOfDay.toLowerCase();
 
-      if (currPeriod === 'morning') {
-        accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr)
-      }
-      if (currPeriod === 'afternoon') {
-        accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr);
-      }
-      if (currPeriod === 'evening') {
-        accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr);
-      }
-      acc[curr.node.date] = accValue;
-      return acc;
-    }, {})
+        if (currPeriod === 'morning') {
+          accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr)
+        }
+        if (currPeriod === 'afternoon') {
+          accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr);
+        }
+        if (currPeriod === 'evening') {
+          accValue[currPeriod] = (accValue[currPeriod] || []).concat(curr);
+        }
+        acc[curr.node.date] = accValue;
+        return acc;
+      }, {})
 
-  return Object.values(mapDataToObject);
+    return Object.values(mapDataToObject);
+  }
```

Comment:
Could we return `null` here if `rawData` is falsy? So that the function always returns something. 
With the current implementation, it'd return undefined.

Reply:
Ok. I would also change it in other places where we used only`return`. That also returns undefined.

Reply:
I have added `null` here. You could change the other ones. Thank you.